File: src-tauri/src/core/git.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct GitStatus {
    pub is_git: bool,
    pub remote_url: Option<String>,
    pub branch: Option<String>,
    pub has_uncommitted_changes: bool,
    pub has_unpushed_commits: bool,
    pub last_sync: Option<u64>,
}

File: internal/models/git.go
package models


type GitFileStatus struct {
	Path   string `yaml:"path" json:"path"`
	Status string `yaml:"status" json:"status"`
}

type GitStatus struct {
	IsGit                 bool            `yaml:"isGit" json:"isGit"`
	RemoteURL             string          `yaml:"remoteUrl" json:"remoteUrl"`
	Branch                string          `yaml:"branch" json:"branch"`
	HasUncommittedChanges bool            `yaml:"hasUncommittedChanges" json:"hasUncommittedChanges"`
	HasUnpushedCommits    bool            `yaml:"hasUnpushedCommits" json:"hasUnpushedCommits"`
	LastSync              int64           `yaml:"lastSync" json:"lastSync"`
	UncommittedFiles      []GitFileStatus `yaml:"uncommittedFiles" json:"uncommittedFiles"`
}

